### ISSUE TITLE
Using toggle fast travel doesn't record map memory at all

### ISSUE THREAD

**Issue by CoroNaut** [CleverRaven/Cataclysm-DDA#76350]
### Describe the bug

When using the toggle fast travel option, it doesn't record map memory for your character. So if you decide to fast travel into an area that you have no map memory for, it will path there, but won't record any tiles.

### Attach save file

[TESTINGTWO-trimmed.tar.gz](https://github.com/user-attachments/files/16960722/TESTINGTWO-trimmed.tar.gz)

### Steps to reproduce

1. Load save
2. Fast travel to any nearby forest tile using 'toggle fast travel'
3. Notice no map memory is updated

### Expected behavior

Map memory should be recorded as if 'toggle fast travel' were off. 

### Screenshots

![image](https://github.com/user-attachments/assets/76cf8de0-621a-4316-a6df-ad36521755b9)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.4651 (22H2)
- Game Version: cdda-experimental-2024-09-10-0339 c73e4c6 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

**Comment by ShnitzelX2** [comment]
So currently, it looks like map memory is only recorded during map drawing, and fast travel mode doesn't call those draw functions. I would expect map memory to be recorded according to what's visible to the in-game player on the map, independent of what's drawn. That would involve extracting _a lot_ of code from the map draw functions.

It's good to identify this, at any rate. Thanks!

**Comment by kevingranade** [comment]
Correct, for performance reasons it is deeply entwined with the draw code so if you don't run most of the draw code it's not going to be able to decide what yours to memorize. Plus if you run that much of the draw code,  you're going to reverse most of the speed gains from not drawing as you go.

I could imagine doing something like adding some code to disable actually drawing always and running just enough code to get a map memory snapshot once every 12 tiles of travel or so.

**Comment by CoroNaut** [comment]
Adding code to update every so often instead of every tile might be OK for moving around in the daytime. Inside a forest would however leave behind some janky map memory as you go. But definitely not at night. #37968 would become a bit more severe. 

Otherwise, its a decent compromise to keep some performance but still update a little map memory for longer distances through not yet recorded areas.

**Comment by RenechCDDA** [comment]
Based on the confirmed behavior and what's been said I'm going to go ahead and mark this confirmed so we at least remember it exists.

**Comment by sparr** [comment]
#77021 seems like a dupe of this, although it gives examples where every 12 tiles is too rare, and even every 4 tiles would leave gaps.
